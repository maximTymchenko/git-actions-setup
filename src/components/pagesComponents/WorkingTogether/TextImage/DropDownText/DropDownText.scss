@import '../../../../../styles/vars';

.drop-down-text {
    width: 100%;
    max-width: 480px;
    margin-bottom: 35px;

    @media screen and (max-width: $tablet) {
        max-width: 720px;
    }

    @media screen and (max-width: $mobile) {
        max-width: 400px;
    }

    &__wrapper {
        border-bottom: 2px solid $secondary;
        margin-bottom: 20px;
    }

    &__header {
        display: flex;
        justify-content: space-between;
        align-items: center;
    }

    &__icon {
        cursor: pointer;

        &--close {
            transform: rotate(45deg);
        }
    }

    &__title {
        font-family: 'OktaNeue-Semibold';
        font-size: 17px;
        font-weight: 600;
        color: $text;
        margin-bottom: 15px;
    }

    &__body {
        margin-bottom: 13px;
        display: flex;
        justify-content: space-between;
        display: none;

        &--show {
            display: -webkit-box;
            display: -ms-flexbox;
            display: flex;
        }
    }

    &__time {
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        font-weight: 400;
        color: $text__body;
        display: block;
    }

    &__rate {
        font-family: 'OktaNeue-Regular';
        font-size: 15px;
        font-weight: 400;
        color: $text;
    }
}
