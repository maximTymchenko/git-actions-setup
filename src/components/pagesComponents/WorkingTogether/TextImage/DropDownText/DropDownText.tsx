import React, { FC, useState } from 'react';
import showIcon from '../../../../../assets/img/workingTogether/show-icon.svg';
import './DropDownText.scss';

const DropDownText: FC = () => {
    const toggleActiveIcon = (index: number) => {
        if (positionsArray.positions[index] === positionsArray.activePosition) {
            return 'drop-down-text__icon drop-down-text__icon--close';
        } else {
            return 'drop-down-text__icon';
        }
    };

    const [positionsArray, setPositionsArray] = useState({
        activePosition: null,
        positions: [
            { id: 0, title: 'Developer' },
            { id: 1, title: 'QA Engineer' },
            { id: 2, title: 'DevOps' },
            { id: 3, title: 'Project Manager' },
        ],
    });

    const toggleShowRate = (index: number) => {
        setPositionsArray({ ...positionsArray, activePosition: positionsArray.positions[index] });
    };

    const toggleActiveClassRate = (index: number) => {
        if (positionsArray.positions[index] === positionsArray.activePosition) {
            return 'drop-down-text__body drop-down-text__body--show';
        } else {
            return 'drop-down-text__body';
        }
    };

    return (
        <div data-testid="drop-down-text" className="drop-down-text">
            {positionsArray.positions.map((pos, index) => (
                <div key={index} className="drop-down-text__wrapper">
                    <div className="drop-down-text__header">
                        <span className="drop-down-text__title">{pos.title}</span>
                        <img
                            onClick={() => toggleShowRate(index)}
                            className={toggleActiveIcon(index)}
                            src={showIcon}
                            alt="Show"
                        />
                    </div>
                    <div className={toggleActiveClassRate(index)}>
                        <div className="drop-down-text__salary">
                            <span className="drop-down-text__time">Hourly</span>
                            <span className="drop-down-text__rate">$60-$150+/hour</span>
                        </div>
                        <div className="drop-down-text__salary">
                            <span className="drop-down-text__time">Part-time</span>
                            <span className="drop-down-text__rate">$1,000-$2,400+/week</span>
                        </div>
                        <div className="drop-down-text__salary">
                            <span className="drop-down-text__time">Full-time</span>
                            <span className="drop-down-text__rate">$2,000-$4,800+/week</span>
                        </div>
                    </div>
                </div>
            ))}
        </div>
    );
};

export default DropDownText;
