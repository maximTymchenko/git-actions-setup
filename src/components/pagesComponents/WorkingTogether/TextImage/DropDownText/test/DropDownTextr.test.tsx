import React from 'react'
import DropDownText from '../DropDownText'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'

it('renders elem correctly', () => {
  const { getByTestId } = render(<DropDownText title="title"></DropDownText>)
  expect(getByTestId('drop-down-text')).toBeTruthy()
})

it('renders without unexpected changes', () => {
  const tree = renderer
    .create(<DropDownText title="title"></DropDownText>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
