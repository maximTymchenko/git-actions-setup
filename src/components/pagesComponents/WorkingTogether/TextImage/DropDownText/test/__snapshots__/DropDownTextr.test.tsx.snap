// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="drop-down-text"
  data-testid="drop-down-text"
>
  <div
    className="drop-down-text__wrapper"
  >
    <div
      className="drop-down-text__header"
    >
      <span
        className="drop-down-text__title"
      >
        Developer
      </span>
      <img
        alt="Show"
        className="drop-down-text__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="drop-down-text__body"
    >
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Hourly
        </span>
        <span
          className="drop-down-text__rate"
        >
          $60-$150+/hour
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Part-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $1,000-$2,400+/week
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Full-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $2,000-$4,800+/week
        </span>
      </div>
    </div>
  </div>
  <div
    className="drop-down-text__wrapper"
  >
    <div
      className="drop-down-text__header"
    >
      <span
        className="drop-down-text__title"
      >
        QA Engineer
      </span>
      <img
        alt="Show"
        className="drop-down-text__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="drop-down-text__body"
    >
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Hourly
        </span>
        <span
          className="drop-down-text__rate"
        >
          $60-$150+/hour
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Part-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $1,000-$2,400+/week
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Full-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $2,000-$4,800+/week
        </span>
      </div>
    </div>
  </div>
  <div
    className="drop-down-text__wrapper"
  >
    <div
      className="drop-down-text__header"
    >
      <span
        className="drop-down-text__title"
      >
        DevOps
      </span>
      <img
        alt="Show"
        className="drop-down-text__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="drop-down-text__body"
    >
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Hourly
        </span>
        <span
          className="drop-down-text__rate"
        >
          $60-$150+/hour
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Part-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $1,000-$2,400+/week
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Full-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $2,000-$4,800+/week
        </span>
      </div>
    </div>
  </div>
  <div
    className="drop-down-text__wrapper"
  >
    <div
      className="drop-down-text__header"
    >
      <span
        className="drop-down-text__title"
      >
        Project Manager
      </span>
      <img
        alt="Show"
        className="drop-down-text__icon"
        onClick={[Function]}
        src="test-file-stub"
      />
    </div>
    <div
      className="drop-down-text__body"
    >
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Hourly
        </span>
        <span
          className="drop-down-text__rate"
        >
          $60-$150+/hour
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Part-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $1,000-$2,400+/week
        </span>
      </div>
      <div
        className="drop-down-text__salary"
      >
        <span
          className="drop-down-text__time"
        >
          Full-time
        </span>
        <span
          className="drop-down-text__rate"
        >
          $2,000-$4,800+/week
        </span>
      </div>
    </div>
  </div>
</div>
`;
